src/solver/conv_hip_implicit_gemm_bwd_v4r1_xdlops.cpp

```diff
@@ -92,15 +92,10 @@ PerformanceImplicitGemmBwdDataV4R1Xdlops::CalculateGemmABlockCopyPerformancePara
             SrcDataPerRead_GemmM = 1;
 
         // calculate threadwise copy size
-        const auto a_data_per_thread_copy =
-            (GemmKPerBlock * GemmMPerBlock * GemmKPACKSize) / BlockSize;
-
-        if(!(a_data_per_thread_copy > 0))
-            MIOPEN_THROW("invalid performance parameter");
-
-        // GemmABlockCopySrcDataPerRead_GemmM also bounded by size of threadwise copy
-        SrcDataPerRead_GemmM = gcd(SrcDataPerRead_GemmM, a_data_per_thread_copy);
+        auto a_data_per_thread_copy =
+            std::max(1, (GemmKPerBlock * GemmMPerBlock * GemmKPACKSize) / BlockSize);
 
+        a_data_per_thread_copy = lcm(a_data_per_thread_copy, SrcDataPerRead_GemmM);
```

Comment:
a_data_per_thread_copy should not be changed here, because its meaning is how many data each thread would copy, and it's decided by A block matrix size and BlockSize. But we can change a_data_per_thread_copy_gemmm (how many data each thread copy along GemmM dimension)

```
const auto a_data_per_thread_copy_gemmm_min = max(GemmMPerBlock / BlockSize, 1); // this is the minimal data a thread need to copy on GemmM dimension, it should be no smaller than 1 or GemmMPerBlock / BlockSize

const auto a_data_per_thread_copy_gemmm = lcm(a_data_per_thread_copy_gemmm_min, SrcDataPerRead_GemmM) // a_data_per_thread_copy_gemmm should be the least common multiple of a_data_per_thread_copy_gemmm_min and SrcDataPerRead_GemmM

if(a_data_per_thread_copy % a_data_per_thread_copy_gemmm != 0)
            MIOPEN_THROW("invalid performance parameter"); // sanity check
```